modules/drivers/bigquery-cloud-sdk/src/metabase/driver/bigquery_cloud_sdk.clj

```diff
@@ -273,17 +284,24 @@
                (rff {:cols fields})
                (-> rows .iterateAll .iterator iterator-seq))))
 
+(defn- ingestion-time-partitioned-table?
+  [table-id]
+  (t2/exists? :model/Field :table_id table-id :name partitioned-time-field-name :database_partitioned true :active true))
+
 (defmethod driver/table-rows-sample :bigquery-cloud-sdk
   [driver {table-name :name, dataset-id :schema :as table} fields rff opts]
   (let [database (table/database table)
         bq-table (get-table database dataset-id table-name)]
-    (if (#{TableDefinition$Type/MATERIALIZED_VIEW TableDefinition$Type/VIEW
-           ;; We couldn't easily test if the following two can show up as
-           ;; tables and if `.list` is supported for them, so they are here
-           ;; to make sure we don't break existing instances.
-           TableDefinition$Type/EXTERNAL TableDefinition$Type/SNAPSHOT}
-         (.. bq-table getDefinition getType))
-      (do (log/debugf "%s.%s is a view, so we cannot use the list API; falling back to regular query"
+    (if (or (#{TableDefinition$Type/MATERIALIZED_VIEW TableDefinition$Type/VIEW
+               ;; We couldn't easily test if the following two can show up as
+               ;; tables and if `.list` is supported for them, so they are here
+               ;; to make sure we don't break existing instances.
+               TableDefinition$Type/EXTERNAL TableDefinition$Type/SNAPSHOT}
+             (.. bq-table getDefinition getType))
+            ;; if the table is partitioned by ingestion time, using .list or .listTableData won't return values for this
+            ;; the _PARTITIONTIME field, so we need to fall back to using sql
+            (ingestion-time-partitioned-table? (:id table)))
+      (do (log/debugf "%s.%s is a view or a table partitioned by ingestion time, so we cannot use the list API; falling back to regular query"
                       dataset-id table-name)
           ((get-method driver/table-rows-sample :sql-jdbc) driver table fields rff opts))
       (sample-table bq-table fields rff))))
```

Comment:
weirdly enough, the `.list` operator of big query can query the table without a filter even though the table requires a partition filter, so we don't need to do anything for range and date partitioned.

But for the time ingestion partition, we want the pseudo columns to be fingerprinted too, so we need to fallback to using an SQL query.